mm2src/coins/utxo/swap_proto_v2_scripts.rs

```diff
@@ -80,3 +80,56 @@ pub fn taker_payment_script(
         .push_opcode(Opcode::OP_ENDIF)
         .into_script()
 }
+
+/// Builds a script for maker payment with immediate refund path
+pub fn maker_payment_script(
+    time_lock: u32,
+    maker_secret_hash: &[u8],
+    taker_secret_hash: &[u8],
+    maker_pub: &Public,
+    taker_pub: &Public,
+) -> Script {
+    let mut builder = Builder::default()
+        .push_opcode(Opcode::OP_IF)
+        .push_bytes(&time_lock.to_le_bytes())
+        .push_opcode(Opcode::OP_CHECKLOCKTIMEVERIFY)
+        .push_opcode(Opcode::OP_DROP)
+        .push_bytes(maker_pub)
+        .push_opcode(Opcode::OP_CHECKSIG)
+        .push_opcode(Opcode::OP_ELSE)
+        .push_opcode(Opcode::OP_IF)
+        .push_opcode(Opcode::OP_SIZE)
+        .push_bytes(&[32])
+        .push_opcode(Opcode::OP_EQUALVERIFY)
+        .push_opcode(Opcode::OP_HASH160);
+
+    if maker_secret_hash.len() == 32 {
+        builder = builder.push_bytes(ripemd160(maker_secret_hash).as_slice());
+    } else {
+        builder = builder.push_bytes(maker_secret_hash);
+    }
+
+    builder = builder
+        .push_opcode(Opcode::OP_EQUALVERIFY)
+        .push_bytes(taker_pub)
+        .push_opcode(Opcode::OP_CHECKSIG)
+        .push_opcode(Opcode::OP_ELSE)
+        .push_opcode(Opcode::OP_SIZE)
+        .push_bytes(&[32])
+        .push_opcode(Opcode::OP_EQUALVERIFY)
+        .push_opcode(Opcode::OP_HASH160);
+
+    if taker_secret_hash.len() == 32 {
+        builder = builder.push_bytes(ripemd160(taker_secret_hash).as_slice());
+    } else {
+        builder = builder.push_bytes(taker_secret_hash);
+    }
+
+    builder
+        .push_opcode(Opcode::OP_EQUALVERIFY)
+        .push_bytes(maker_pub)
```

Comment:
Maybe not related to this PR, but why do we need this push_bytes()? Seems other apps do not have such a method but they have push_data()-like method (which does push_bytes job as well).
I guess push_bytes is used to prevent sending arrays longer than 75b but not sure why this could be needed.
btw there is a standard limitation for push_data in the stack of 520b

Reply:
mm2_bitcoin crate code was initially borrowed from parity-bitcoin implementation, we also got these 2 fns from there. I'm not sure what was the purpose, actually :slightly_smiling_face: So yes, `push_bytes` can be removed in favor of `push_data`. Will do shortly.

Reply:
Done